sdk/highlight-node/src/client.ts

```diff
@@ -17,14 +15,47 @@ import { Resource, processDetectorSync } from '@opentelemetry/resources'
 
 const OTLP_HTTP = 'https://otel.highlight.io:4318'
 
+// @ts-ignore
+class CustomSpanProcessor extends BatchSpanProcessorBase<BufferConfig> {
+	private _listeners: Map<Symbol, () => void>
+
+	constructor(config: any) {
+		super(config)
+
+		this._listeners = new Map()
+	}
+
+	onShutdown(): void {}
+
+	async forceFlush(): Promise<void> {
+		// @ts-ignore
+		const finishedSpansCount = this._finishedSpans.length
+
+		await super.forceFlush()
+
+		if (finishedSpansCount > 0) {
+			Array.from(this._listeners.values()).forEach((listener) =>
```

Comment:
Call all registered listeners after spans have been flushed successfully. Most `forceFlush` calls don't actually flush any spans, so waiting for a real, live span to pass through is critical.